crates/protocol/src/batch/tx_data/wrapper.rs

```diff
@@ -34,6 +36,10 @@ impl Encodable for SpanBatchTransactionData {
                 out.put_u8(TxType::Eip1559 as u8);
                 data.encode(out);
             }
+            Self::Eip7702(data) => {
+                out.put_u8(TxType::Eip1559 as u8);
```

Comment:
We need the `Eip7702` transaction type here. See [`alloy_consensus::TxType`](https://docs.rs/alloy-consensus/0.9.2/alloy_consensus/transaction/enum.TxType.html).

```suggestion
                out.put_u8(TxType::Eip7702 as u8);
```